### ISSUE TITLE
Mismatch between self-exported yolov5s.onnx model and officially released yolov5s.onnx

### ISSUE THREAD

**Issue by August0424** [ultralytics/yolov5#2652]
## I exported the official released yolov5s.pt to onnx, whose network structure visualized by Netron differs from that of officially exported yolov5s.onnx

I downloaded yolo5s.onnx from https://github.com/ultralytics/yolov5/releases/download/v4.0/yolov5s.onnx, following the instructions of https://github.com/OAID/Tengine/tree/tengine-lite/tools/optimize to deploy YOLOv5s on NPU of edge device. This officially exported onnx model works in this course, therefore I decided to finetune an official provided yolov5s.pt from https://github.com/ultralytics/yolov5/releases/download/v4.0/yolov5s.pt and then manually export it to onnx format (using models/export.py) for the same deployment.

However, my self-exported yolov5s.onnx failed to detect object when deployed in the same manner, though the code runned successfully. After that, I carefully examined the difference of network struture between the 2 onnx models using Netron, and found that their topology and size of feature maps are approximately the same, except for that the model downloaded from https://github.com/ultralytics/yolov5/releases/download/v4.0/yolov5s.onnx outputs 3 vectors of 2-dimensions (ignoring the batch size dim) like (25560, 85), wheras my onnx model (mannully exported from pretrained yolov5s.pt) ends with 3 4-dim vectors like (3, 80, 80, 85). I don't remember the exact number, but acturally they can be reshaped from each other since their size are the same. It is likely that this minor difference had led to the failture of detection of my own model in deployment environment.

I just wonder why I couldn't reproduce the structure of downloaded yolov5s.onnx by exporting .pt to onnx using models/export.py ? Could you please tell me how the onnx model  (https://github.com/ultralytics/yolov5/releases/download/v4.0/yolov5s.onnx) can be generated from an officially .pt model ? Looking forward to your quick reply !

## Plus
Here is the screenshot of the 2 onnx model of yolov5s mentioned above. The left one corresponds to the onnx model directly downloaded from https://github.com/ultralytics/yolov5/releases/download/v4.0/yolov5s.onnx, which outputs 3 2-dimentional vectors. And the right counterpart refers to the onnx mannully exported from a pretrained yolov5s.pt, which outputs 3 4-dimantional vectors.
![图片](https://user-images.githubusercontent.com/54147016/112923733-60ed5380-9141-11eb-8820-48b023018a8e.png)
